x/exchange/keeper/keeper.go

```diff
@@ -0,0 +1,270 @@
+package keeper
+
+import (
+	"fmt"
+	"strings"
+
+	"github.com/gogo/protobuf/proto"
+
+	sdkmath "cosmossdk.io/math"
+
+	"github.com/cosmos/cosmos-sdk/codec"
+	"github.com/cosmos/cosmos-sdk/store/prefix"
+	storetypes "github.com/cosmos/cosmos-sdk/store/types"
+	sdk "github.com/cosmos/cosmos-sdk/types"
+	authtypes "github.com/cosmos/cosmos-sdk/x/auth/types"
+	banktypes "github.com/cosmos/cosmos-sdk/x/bank/types"
+	govtypes "github.com/cosmos/cosmos-sdk/x/gov/types"
+	"github.com/cosmos/cosmos-sdk/x/quarantine"
+
+	"github.com/provenance-io/provenance/x/exchange"
+)
+
+var (
+	// OneInt is an sdkmath.Int of 1.
+	OneInt = sdkmath.NewInt(1)
+	// TenKInt is an sdkmath.Int of 10,000.
+	TenKInt = sdkmath.NewInt(10_000)
```

Comment:
Is there a stronger name that we can use? I'm asking because TenKInt is not really that much different than just using sdkmath.NewInt(10_000), and I had trouble deciphering what was going on at first:

`splitAmt, splitRem := exchange.QuoRemInt(coin.Amount.Mul(sdkmath.NewInt(split)), TenKInt)`

Reply:
There's no need to recreate the exact same thing over and over again, specially when it's immutable. Calling a function (e.g. `NewInt`) requires more resources than accessing a variable (even when the function is just returning a variable (e.g. `GetAmount() string { return f.amount }`)).

Yes, the 10,000 is for converting from basis points to whatever it's called as an actual value. Usually, you'd think of it as amount * (split/10,000), but with integer math, we need to do that multiplication before the division.

As it is the line is, "give me the quotient and remainder when I multiply the amount times the split and divide it by 10,000." Calling the variable anything other than "ten thousand" would make that line less readable because you then don't actually know what math is being done; "what am I dividing by here?"

`QuoRem` is a function that `math.BigInt` has, but the sdk didn't see fit to also have that method for an `sdkmath.Int`, which is why I had to make that `exchange.QuoRemInt` function.